File: app/src/test/java/com/dougkeen/bart/networktasks/NetworkUtilsTest.java
package com.dougkeen.bart.networktasks;

import static org.junit.Assert.assertEquals;

import okhttp3.OkHttpClient;

import org.junit.Test;

public class NetworkUtilsTest {
    @Test
    public void clientUsesBoundedTimeouts() {
        OkHttpClient client = NetworkUtils.makeHttpClient();

        assertEquals(NetworkUtils.CONNECTION_TIMEOUT_MILLIS,
                client.connectTimeoutMillis());
        assertEquals(NetworkUtils.READ_TIMEOUT_MILLIS,
                client.readTimeoutMillis());
        assertEquals(NetworkUtils.WRITE_TIMEOUT_MILLIS,
                client.writeTimeoutMillis());
        assertEquals(NetworkUtils.CALL_TIMEOUT_MILLIS,
                client.callTimeoutMillis());
    }
}

File: app/src/test/kotlin/com/dougkeen/bart/networktasks/NetworkUtilsTest.kt
package com.dougkeen.bart.networktasks

import okhttp3.OkHttpClient
import org.junit.Assert.assertEquals
import org.junit.Test

class NetworkUtilsTest {
    @Test
    fun clientUsesBoundedTimeouts() {
        val client: OkHttpClient = NetworkUtils.makeHttpClient()
        assertEquals(NetworkUtils.CONNECTION_TIMEOUT_MILLIS, client.connectTimeoutMillis)
        assertEquals(NetworkUtils.READ_TIMEOUT_MILLIS, client.readTimeoutMillis)
        assertEquals(NetworkUtils.WRITE_TIMEOUT_MILLIS, client.writeTimeoutMillis)
        assertEquals(NetworkUtils.CALL_TIMEOUT_MILLIS, client.callTimeoutMillis)
    }
}
